File: app/src/main/java/com/example/firabasefirstexperience/java/adapter/MessageAdapter.java
package com.example.firabasefirstexperience.java.adapter;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.firabasefirstexperience.java.model.Chat;
import com.example.firabasefirstexperience.R;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

import java.util.List;

public class MessageAdapter extends RecyclerView.Adapter<MessageAdapter.ViewHolder> {


    public static final int MSG_LEFT = 0;
    public static final int MSG_RIGHT = 1;

    private Context mContext;
    private List<Chat> listChats;

    public MessageAdapter(Context mContext, List<Chat> listChats) {
        this.mContext = mContext;
        this.listChats = listChats;
    }


    @NonNull
    @Override
    public MessageAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        if (i == MSG_RIGHT) {
            View view = LayoutInflater.from(mContext).inflate(R.layout.chat_item_right, viewGroup, false);
            return new MessageAdapter.ViewHolder(view);
        } else {
            View view = LayoutInflater.from(mContext).inflate(R.layout.chat_item_left, viewGroup, false);
            return new MessageAdapter.ViewHolder(view);
        }
    }

    @Override
    public void onBindViewHolder(@NonNull MessageAdapter.ViewHolder viewHolder, int i) {

        viewHolder.bind(listChats.get(i));

    }

    @Override
    public int getItemCount() {
        return listChats.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        private TextView tv_showMessage;


        ViewHolder(View itemView) {
            super(itemView);

            tv_showMessage = itemView.findViewById(R.id.tv_showMessage);
        }


        void bind(Chat chat) {
            tv_showMessage.setText(chat.getMessage());
        }


    }

    @Override
    public int getItemViewType(int position) {
        FirebaseUser fUser = FirebaseAuth.getInstance().getCurrentUser();

        if (listChats.get(position).getSender().equals(fUser.getUid())) {
            return MSG_RIGHT;
        } else {
            return MSG_LEFT;
        }
    }
}

File: app/src/main/java/com/example/firabasefirstexperience/kotlin/presentation/chat/MessageAdapter.kt
package com.example.firabasefirstexperience.kotlin.presentation.chat

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.firabasefirstexperience.R
import com.example.firabasefirstexperience.kotlin.model.Chat
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser

class MessageAdapter(private var listChats: List<Chat>) : RecyclerView.Adapter<MessageAdapter.ViewHolder>() {

    private var MSG_LEFT = 0
    private var MSG_RIGHT = 1

    override fun onCreateViewHolder(parent: ViewGroup, postion: Int): ViewHolder {
        if (postion == MSG_LEFT) {
            val view: View = LayoutInflater.from(parent.context).inflate(R.layout.chat_item_left, parent, false)
            return ViewHolder(view)
        } else {
            val view: View = LayoutInflater.from(parent.context).inflate(R.layout.chat_item_right, parent, false)
            return ViewHolder(view)
        }
    }

    override fun getItemCount(): Int {
        return listChats.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(listChats[position])
    }


    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        private var tv_showMessge: TextView = itemView.findViewById(R.id.tv_showMessage)

        fun bind(chat: Chat) {
            tv_showMessge.text = chat.message
        }
    }

    override fun getItemViewType(position: Int): Int {
        val fbUser: FirebaseUser? = FirebaseAuth.getInstance().currentUser

        return if (listChats[position].sender == (fbUser?.uid)) {
            MSG_RIGHT
        } else MSG_LEFT

    }
}